File: SDKs/amazonaws.com/rekognition/2016-06-27/java/lib/src/main/java/openapisdk/models/shared/UnindexedFace.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * UnindexedFace
 * A face that <a>IndexFaces</a> detected, but didn't index. Use the <code>Reasons</code> response attribute to determine why a face wasn't indexed.
**/
public class UnindexedFace {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("FaceDetail")
    public FaceDetail faceDetail;
    public UnindexedFace withFaceDetail(FaceDetail faceDetail) {
        this.faceDetail = faceDetail;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("Reasons")
    public ReasonEnum[] reasons;
    public UnindexedFace withReasons(ReasonEnum[] reasons) {
        this.reasons = reasons;
        return this;
    }
}

File: SDKs/amazonaws.com/rekognition/2016-06-27/python/src/sdk/models/shared/unindexedface.py
import dataclasses
from typing import Optional
from enum import Enum
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import facedetail as shared_facedetail
from ..shared import reason_enum as shared_reason_enum


@dataclass_json
@dataclasses.dataclass
class UnindexedFace:
    r"""UnindexedFace
    A face that <a>IndexFaces</a> detected, but didn't index. Use the <code>Reasons</code> response attribute to determine why a face wasn't indexed.
    """
    
    face_detail: Optional[shared_facedetail.FaceDetail] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('FaceDetail') }})
    reasons: Optional[list[shared_reason_enum.ReasonEnum]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('Reasons') }})
    
